src/dissect/dissect.c

```diff
@@ -353,17 +339,17 @@ static int action_dissect(DissectedImage *m, LoopDevice *d) {
         assert(m);
         assert(d);
 
-        if (!arg_json)
+        if (arg_json_format_flags & JSON_FORMAT_OFF)
```

Comment:
nit: when are we supposed to use FLAGS_SET and when not? It seems a bit incosistent through the code base

Reply:
FLAGS_SET() matters in particular when you test flags that are combinations of bits, or multiple flags at once. For the rest it's up to you I guess. I tend to use the syntax `if (x & flag)` for simple flag check, but when checking if a bit is off i use `if (!FLAGS_SET(x, flag))` right now, since i found `if (!(x & flag))` so ugly, because of the `(!(` part, but I don't expect anyone to agree with me on that... ;-)

basically, it's up to you, but if you spell out `x & (f1|f2) == (f1|f2)` then you better use `FLAGS_SET(x, f1|f2)`.

the macro is called "FLAG*S*" after all, i.e. is in plural, pointing to the fact that it is useful when checking multiple flags at once